### DISCUSSION TITLE
npm Category Guidelines 🗃️

### DISCUSSION THREAD

**Discussion by grandmas-favorite** [community/community#121825] [npm]
**Got thoughts about npm? The cli, documentation, the registry, etc? Here’s the place to chat! We welcome your feedback and conversations, so feel free to start a discussion!**

✅ What **is allowed** in this category:
- Technical questions about npm, the cli, documentation, registry, etc.
- Product feedback requests
- Conversations and opinions on tooling/packaging

 ❌What **is not allowed** in this category:
- Discussions regarding GitHub Packages (will be recategorized)
- Posting job listings or contests
- Self-promoting your projects

If you post something that is **not allowed**, your post may be removed without notice and you may receive a temporary ban 🛑.

If you’re interested in learning more about this category, we’ve put together a FAQ below ❔. Have unanswered questions? Please let us know in the comments below.

**Comment by JorenVsyntra** [comment]
Hey everyone!

**Comment by jerryTomson298** [comment]
It sounds like a great space for developers to discuss everything related to npm! The CLI, documentation, and registry are core parts of the JavaScript ecosystem, so having a dedicated place for feedback and discussion can be really helpful.

The inclusion of product feedback requests is a plus—it gives users a way to directly influence npm's development. However, the restrictions on self-promotion and off-topic discussions make sense to keep things focused.

One thing that could be interesting is how npm plans to engage with user feedback. Will the npm team be active in responding to issues, or is this more of a community-driven space? Also, how does it compare to discussions happening on GitHub Issues or other forums like Stack Overflow?

Have you participated in discussions there yet? I'd love to hear what kind of feedback people are sharing!

**Reply by rishadmohammad2007-ai** [reply]
> It sounds like a great space for developers to discuss everything related to npm! The CLI, documentation, and registry are core parts of the JavaScript ecosystem, so having a dedicated place for feedback and
[firebase_task_app.html](https://github.com/user-attachments/files/24788737/firebase_task_app.html)
[icon-zip-manager.tsx.txt](https://github.com/user-attachments/files/24788747/icon-zip-manager.tsx.txt)
 discussion can be really helpful.
> 
> The inclusion of product feedback requests is a plus—it gives users a way to directly influence npm's development. However, the restrictions on self-promotion and off-topic discussions make sense to keep things focused.
> 
> One thing that could be interesting is how npm plans to engage with user feedback. Will the npm team be active in responding to issues, or is this more of a community-driven space? Also, how does it compare to discussions happening on GitHub Issues or other forums like Stack Overflow?
> 
> Have you participated in discussions there yet? I'd love to hear what kind of feedback people are sharing!

**Comment by asma019** [comment]
Thanks for putting these guidelines together! 📝
It’s really helpful to have clear do’s ✅ and don’ts ❌ so discussions stay focused and productive. I like that the category is open to technical questions, product feedback, and deeper conversations about npm tooling — it encourages more collaborative problem-solving.

Also appreciate the clarity on what’s not allowed, especially around self-promotion and off-topic posts, since it keeps the space signal-heavy and spam-free.

Looking forward to contributing useful feedback and learning from others here! 🚀

**Comment by wolfletexed** [comment]
Hello npm Team,

I hope this message finds you well. As a developer who often works at night, I wanted to share a suggestion that I believe would greatly improve the user experience: adding a dark mode option to your website.

**Reply by amanmohdtp** [reply]
**That's real. Like dark mode is really way more compatible.**

**Comment by HAMZA0622** [comment]
I completely agree with the dark mode suggestion. Many developers spend long hours working in low-light environments, and a native dark theme would make the npm website more comfortable to use while reducing eye strain.

It would also help create a more consistent experience with modern developer tools and platforms that already support dark mode. Hopefully, the npm team considers adding it in a future update.

Thanks for bringing up this idea!
